plugin/trino-singlestore/src/test/java/io/trino/plugin/singlestore/TestSingleStorePlugin.java

```diff
@@ -45,4 +49,24 @@ public void testCreateConnectorLegacyName()
                 .orElseThrow();
         factory.create("test", ImmutableMap.of("connection-url", "jdbc:singlestore://test"), new TestingConnectorContext()).shutdown();
     }
+
+    @Test
+    public void testLegacyConnectionUrl()
```

Comment:
This seems redundant over the test below - one tests the method directly - the other tests user observable (i.e. production code-path).

Reply:
I'd argue, strongly, that both are required and necessary.
One tests the method (unit test), the other tests the actual production code-path.
My biggest issue with Guice/Injection is that it allows writing untestable code (objects created by field injection, validation that occurs via annotations, etc, etc), code that can only be tested by creating the Injector, etc.